crypto11/ecdsa.go

```diff
@@ -151,7 +151,7 @@ func marshalEcParams(c elliptic.Curve) ([]byte, error) {
 func unmarshalEcParams(b []byte) (elliptic.Curve, error) {
 	// See if it's a well-known curve
 	for _, ci := range wellKnownCurves {
-		if bytes.Compare(b, ci.oid) == 0 {
+		if bytes.Equal(b, ci.oid) {
```

Comment:
This implies to me we're adding a different version of crypto11 than what's in production.

Could we get the same version incorporated? Or should we upgrade the vendored version first?

Reply:
The reason this is getting picked up now is because it's no longer in the `vendor` directory, AIUI. I was trying to avoid any added risk by upgrading, but I can give it a shot if you like.

Reply:
(Or we can just ignore all the new linting failures and use exactly the same code as we are now.)

Reply:
I'm sorry? No, I'm saying the code that's in vendor isn't the same that's being added here in the new directory. I'm trying to say you're adding a different version than what's in our vendor directory right now and I would like to either not do that. 

I suspected this was a newer version but it could be this crypto11 is of an older version. In either case, I'd like for us to be exact in what we're doing

Reply:
I'm pretty sure I'm doing this? https://github.com/mozilla-services/autograph/pull/984/commits/5432a9051bd49487dbd54e8372646cc4c63ab1bd moves all of the crypto11 stuff from `vendor` up to the new `crypto11` directory - note the newly failing linting tests on that, despite no actual changes to the code.

https://github.com/mozilla-services/autograph/pull/984/commits/23dfeb96b4cdbf91c8a718b55ddcf6a33c1dd9a1 then makes the minor changes you're seeing to fix the linting issues.

Reply:
Oh, I see. I missed that there were changes because of our linting.